iroh-gateway/src/config.rs

```diff
@@ -2,122 +2,167 @@ use std::net::SocketAddr;
-    /// set of user provided headers to attach to all responses
-    pub headers: HeaderMap,
     /// default port to listen on
     pub port: u16,
-    pub rpc: RpcConfig,
-}
-
-#[derive(Debug, Clone)]
-pub struct RpcConfig {
-    /// Address on which to listen,
-    pub listen_addr: SocketAddr,
-    pub client_config: RpcClientConfig,
-}
-
-impl Default for RpcConfig {
-    fn default() -> Self {
-        let client_config = RpcClientConfig::default();
-        RpcConfig {
-            listen_addr: client_config.gateway_addr,
-            client_config,
-        }
-    }
+    /// rpc listening addr
+    pub rpc_addr: SocketAddr,
+    // NOTE: for toml to serialize properly, the "table" values must be serialized at the end, and
+    // so much come at the end of the `Config` struct
+    /// set of user provided headers to attach to all responses
+    #[serde(with = "http_serde::header_map")]
+    pub headers: HeaderMap,
+    /// rpc addresses for the gateway & addresses for the rpc client to dial
+    pub rpc_client: RpcClientConfig,
 }
 
 impl Config {
-    pub fn new(writeable: bool, fetch: bool, cache: bool, port: u16, rpc: RpcConfig) -> Self {
+    pub fn new(
+        writeable: bool,
+        fetch: bool,
+        cache: bool,
+        port: u16,
+        rpc_addr: SocketAddr,
+        rpc_client: RpcClientConfig,
+    ) -> Self {
         Self {
             writeable,
             fetch,
             cache,
             headers: HeaderMap::new(),
             port,
-            rpc,
+            rpc_addr,
+            rpc_client,
         }
     }
 
     pub fn set_default_headers(&mut self) {
-        let mut headers = HeaderMap::new();
-        headers.typed_insert(AccessControlAllowOrigin::ANY);
-        headers.typed_insert(
-            [
-                Method::GET,
-                Method::PUT,
-                Method::POST,
-                Method::DELETE,
-                Method::HEAD,
-                Method::OPTIONS,
-            ]
-            .into_iter()
-            .collect::<AccessControlAllowMethods>(),
-        );
-        headers.typed_insert(
-            [
-                CONTENT_TYPE,
-                CONTENT_DISPOSITION,
-                LAST_MODIFIED,
-                CACHE_CONTROL,
-                ACCEPT_RANGES,
-                ETAG,
-                HEADER_SERVICE_WORKER.clone(),
-                HEADER_X_IPFS_GATEWAY_PREFIX.clone(),
-                HEADER_X_TRACE_ID.clone(),
-                HEADER_X_CONTENT_TYPE_OPTIONS.clone(),
-                HEADER_X_IPFS_PATH.clone(),
-                HEADER_X_IPFS_ROOTS.clone(),
-            ]
-            .into_iter()
-            .collect::<AccessControlAllowHeaders>(),
-        );
-        // todo(arqu): remove these once propperly implmented
-        headers.insert(CACHE_CONTROL, VALUE_NO_CACHE_NO_TRANSFORM.clone());
-        headers.insert(ACCEPT_RANGES, VALUE_NONE.clone());
-        self.headers = headers;
+        self.headers = default_headers();
```

Comment:
needed to test the conversion between `config::Config` and the `http::HeaderMap`, so split this out so it could be used to help with tests separate from the `gateway::Config`